regolith/filter_nim.go

```diff
@@ -101,19 +115,31 @@ func (f *NimFilterDefinition) InstallDependencies(
 		installLocation = parent.GetDownloadPath(dotRegolithPath)
 	}
 	Logger.Infof("Downloading dependencies for %s...", f.Id)
-	joinedPath := filepath.Join(installLocation, f.Script)
-	scriptPath, err := filepath.Abs(joinedPath)
-	if err != nil {
-		return WrapErrorf(err, filepathAbsError, joinedPath)
+	var requirementsPath string
+	if f.Requirements == "" {
+		// Deduce the path from the script path
+		joinedPath := filepath.Join(installLocation, f.Script)
+		scriptPath, err := filepath.Abs(joinedPath)
+		if err != nil {
+			return WrapErrorf(err, filepathAbsError, joinedPath)
+		}
+		requirementsPath = filepath.Dir(scriptPath)
+	} else {
+		joinedPath := filepath.Join(installLocation, f.Requirements)
+		scriptPath, err := filepath.Abs(joinedPath)
+		if err != nil {
+			return WrapErrorf(err, filepathAbsError, joinedPath)
+		}
+		requirementsPath = scriptPath
 	}
-	filterPath := filepath.Dir(scriptPath)
-	if hasNimble(filterPath) {
+	Logger.Debugf("Installing dependencies using nimble in %s", requirementsPath)
```

Comment:
Can you justify why you chose to do a directory path, instead of the full text of the file? I'm leaving comment here, but also applies to Python etc. 

Wouldn't it make more sense to specify `./filters/whatever/requirements.txt` rather than just the directory containing the requirements file?

Reply:
Python is the only case where you can do that. Nodejs and nim uses path to the directory. I could change that only for python but it would be inconsistent.

Reply:
Sorry, didn't know that. 

But your logic is flawed. Python does NOT allow specifying path to the directory. It requires a file. And the file could be called `requirements.txt` or `sdfjsdjfdsfsd.txt` it doesn't matter. 

The python version should be adjusted to take the full path to the requirements file, otherwise it would block some valid setups. 

The convention is to use `requirements.txt` but you can find requirements files with other names quite often.

Reply:
https://pip.pypa.io/en/stable/reference/requirements-file-format/

Reply:
I'm aware that you can name your file differently but naming it `requirements.txt` is the most common and best practice. I'm not sure if sacrificing consistency, just to give the user the option to do something that might be considered bad practice is a good idea.

Reply:
For me it's fairly important that we keep Regoliths doors open to additional projects, which weren't intended to run on Regolith. Keeping the doors open for arbitrary requirements files would be a nice nod to that.